app/src/main/java/eu/kanade/tachiyomi/data/backup/restore/restorers/ExtensionRepoRestorer.kt

```diff
@@ -0,0 +1,39 @@
+package eu.kanade.tachiyomi.data.backup.restore.restorers
+
+import eu.kanade.tachiyomi.data.backup.models.BackupExtensionRepos
+import mihon.domain.extensionrepo.interactor.GetExtensionRepo
+import tachiyomi.data.DatabaseHandler
+import uy.kohesive.injekt.Injekt
+import uy.kohesive.injekt.api.get
+
+class ExtensionRepoRestorer(
+    private val handler: DatabaseHandler = Injekt.get(),
+    private val getExtensionRepos: GetExtensionRepo = Injekt.get()
+) {
+
+    suspend operator fun invoke(backupExtensionRepos: List<BackupExtensionRepos>) {
+        if (backupExtensionRepos.isEmpty()) return
+
+        val dbExtensionRepos = getExtensionRepos.getAll()
+        val dbExtensionReposBySHA = dbExtensionRepos.associateBy { it.signingKeyFingerprint }
+
+        backupExtensionRepos
+            .sortedBy { it.baseUrl }
+            .forEach { backupRepo ->
+                val dbExtensionRepo = dbExtensionReposBySHA[backupRepo.signingKeyFingerprint]
+                if (dbExtensionRepo == null) {
```

Comment:
Get by url. if exists check fingerprint match if doesn't match pass error. if doesn't exist then check if some other repo has the fingerprint. if they do then pass an errro. otherwise restore. also do a try catch to catch any other general error inside the loop and pass it via lambda